tests/tests_basic.yml

```diff
@@ -6,33 +6,127 @@
+            group: root
+          loop: [httpd1, httpd2]
 
-    - name: Check if containers are running
-      command: podman ps -a
-      changed_when: false
+        - name: Create data files
+          copy:
+            dest: /tmp/{{ item }}/index.txt
+            content: "123"
+            mode: "0644"
+            owner: root
+            group: root
+          loop: [httpd1, httpd2]
 
-    - name: Run role again to test for idempotency
-      include_role:
-        name: linux-system-roles.podman
+        - name: Run role
+          include_role:
+            name: linux-system-roles.podman
+
+        - name: Check if containers are running
+          command: podman inspect {{ item }} --format '{{ __fmt }}'
+          changed_when: false
+          register: __output
+          failed_when: __output.stdout != 'running'
+          vars:
+            __fmt: "{{ '{' ~ '{.State.Status}' ~ '}' }}"
+          loop: [httpd1, httpd2]
+
+        - name: Get service facts
+          service_facts:
+
+        - name: Check services
+          assert:
+            that: __state == 'running'
+          vars:
+            __srv: "{{ ansible_facts.services.keys() |
+              select('match', '^lsr_container-' ~ item ~ '.service') |
+              first }}"
+            __state: "{{ ansible_facts.services[__srv]['state'] }}"
+          loop: [httpd1, httpd2]
+
+        - name: Check ports, data
+          uri:
+            url: http://localhost:{{ item }}/index.txt
+            return_content: true
+          register: __return
+          failed_when: __return.content != "123"
+          loop: [8080, 8081]
+
+        - name: Run role again to test for idempotency
+          include_role:
+            name: linux-system-roles.podman
+
+        - name: Get service facts
+          service_facts:
+
+        - name: Check services
+          assert:
+            that: __state == 'running'
+          vars:
+            __srv: "{{ ansible_facts.services.keys() |
+              select('match', '^lsr_container-' ~ item ~ '.service') |
+              first }}"
+            __state: "{{ ansible_facts.services[__srv]['state'] }}"
+          loop: [httpd1, httpd2]
+
+        - name: Check ports, data
+          uri:
+            url: http://localhost:{{ item }}/index.txt
+            return_content: true
+          register: __return
+          failed_when: __return.content != "123"
+          loop: [8080, 8081]
+      always:
+        # have to clean up storage.conf - otherwise, get this message:
+        # A storage.conf file exists at /etc/containers/storage.conf
+        # You should remove this file if you did not modify the configuration.
+        - name: Clean up storage.conf
+          file:
+            path: /etc/containers/storage.conf
+            state: absent
+          tags:
+            - tests::cleanup
+
+        - name: Clean up all containers and associated systemd services
+          command: podman system reset --force
```

Comment:
Note that it does not clean up systemd services.  It's an interesting idea though!